src/components/button/button.spec.tsx

```diff
@@ -0,0 +1,31 @@
+import Button from "./button"
+import { ButtonType } from '../constants/button-types'
+import { render, screen } from '@testing-library/react'
+import { faker } from '@faker-js/faker'
+
+describe('Button Component', () => {
+  test('Should render correctly', () => {
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} />)
+    const button = screen.getByRole('button-' + ButtonType.Success)
+    expect(button).toBeInTheDocument()
+  })
+
+  test('Should call on click', () => {
+    const actionClicked = jest.fn()
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} onClick={actionClicked} />)
+    const button = screen.getByRole('button-' + ButtonType.Success)
+    button.click()
+    expect(actionClicked).toHaveBeenCalledTimes(1)
+  })
+
+  test('Should display only one variant', () => {
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} />)
```

Comment:
Não é uma boa prática chamar dessa forma, sugiro externalizar para uma constante
```suggestion
    const label = faker.lorem.word()
    render(<Button variant={ButtonType.Success} label={label} />)
```

Reply:
commit: 85a3f047bf401c3014feb43e4c2bb42233b3ab15